### PR TITLE
Add support for building & testing Sysbox on Amazon Linux 2 hosts.

### PR DESCRIPTION
**Pull request by ctalledo**

### CODE REVIEW

File: tests/buildx/basic.bats [hunk 1]
```diff
@@ -58,6 +58,11 @@ EOF
 # buildkit runs at L1 level).
 @test "L0 buildx + L1 buildkit container (docker-container driver)" {
 
+  docker buildx version
+  if [ "$status" -ne 0 ]; then
```
Comment (rodnymolina) [inline tests/buildx/basic.bats:62]: If buildx is not supported then we're probably dealing with a relatively-old docker version, which makes me wonder if big improvements like [this](https://github.com/nestybox/sysbox/blob/master/docs/user-guide/configuration.md#speeding-up-sysbox-by-disallowing-trusted-overlay-xattributes) one is supported in amazon-linux.

File: tests/pods/basic.bats [hunk 2]
```diff
@@ -10,6 +10,13 @@ load ../helpers/k8s
 load ../helpers/run
 load ../helpers/sysbox-health
 
+function setup() {
```
Comment (rodnymolina) [inline tests/pods/basic.bats:13]: Why are you doing this? There's no option to install cri-o in amazon-linux?

**Review by rodnymolina — APPROVED** [review]
LG

File: tests/buildx/basic.bats (hunk 1 above)
Comment (ctalledo) [reply tests/buildx/basic.bats:62]: It's not that old; Amazon Linux comes with Docker 20.10, but it does not support `docker buildx`. I tried installing the latest Docker following the official Docker installation instructions, but it's not supported on AL2.

File: tests/pods/basic.bats (hunk 2 above)
Comment (ctalledo) [reply tests/pods/basic.bats:13]: No, the CRI-O installation instructions don't support AL2. We would need to build from source if we want to run CRI-O in the test container on AL2 (in a future commit).

**Pull request merged**